Parser/tokenizer.c

```diff
@@ -370,6 +370,57 @@ update_fstring_buffers(struct tok_state *tok, char value, int regular, int multi
     }
 }
 
+static int
+update_fstring_expr(struct tok_state *tok, char cur)
+{
+    if (!tok->cur || !tok->tok_mode_stack_index) {
+        return 1;
+    }
```

Comment:
Maybe we should make this an assert and move the check to the caller for better readability (someone reading the caller could ask "why we are updating the fstring if we may not be in an f-string"). I am not feeling strongly though

Reply:
We could possibly inline this in the 2 callsites that call this function without checking and assert here.